vars/piperPipelineStageBuild.groovy

```diff
@@ -54,7 +56,7 @@ void call(Map parameters = [:]) {
 
         durationMeasure(script: script, measurementName: 'build_duration') {
 
-            buildExecute script: script
+            buildExecute script: script, buildTool: parameters.buildTool
```

Comment:
This looks strange. You define the key in GENERAL_CONFIG_KEYS, not in PARAMETER_KEYS, but then use it from parameters, not the resolved config. Maybe you just need to declare it in PARAMETER_KEYS instead?